openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,850 @@
-                        tempCImax = Math.max.apply(null, tempCIvalues);
-
-                    // Scale the values
-                    for (var j = 0; j < tempCIvalues.length; j++) {
-                        scaleCIvalues.push( (tempCIvalues[j] - tempCImin) / (tempCImax - tempCImin) );
-                    }
-
-                    // Call the applyScaledValues method
-                    for (var l in scaledCatIndicator) {
-                        scaledCatIndicator[l][CIkeys[i]] = scaledCatIndicator[l].scaleCIvalues(l);
+                var munic3 = la[o].properties[selectedRegion];
+                var theme3 = name;
+                tempString.push(munic3 + '|'+ theme3 +'|'+ tempSum);
+            } else if ( operator == "Weighted multiplication") {
+                for (var p4 in la[o].properties) {
+                    // iterate over the indicator child keys
+                    for (var r4 = 0; r4 < indicatorChildrenKey.length; r4++, ct++) {
+                        if (p4 == indicatorChildrenKey[r4]) {
+                            // Sum the theme indicators
+                            var weight = tempChildren[r4].weight;
+                            if (tempSum == 0) {
+                                tempSum = (la[o].properties[p4] * weight);
+                            } else {
+                                tempSum = tempSum * (la[o].properties[p4] * weight);
+                            }
+                        }
                     }
                 }
-                scaleCIvalues = [];
-                tempCIvalues = [];
-            }
-
-            // The data passed into d3 need to be an array of objects
-            // Place the category indicator objects into an array
-            var catData = [];
-
-            for (var k in scaledCatIndicator) {
-                catData.push(scaledCatIndicator[k]);
-            }
-
-            // TODO remove this concat if need be
-            var concat = catData.concat(previousCatData);
-
-            Category_PCP_Chart(catData, municipality, districName, concat);
-
-            for (var k in scaledCatIndicator) {
-                previousCatData.push(scaledCatIndicator[k]);
-            }
-
-            /////////////////////////////////////////////
-            /////////// Create the svi object ///////////
-            /////////////////////////////////////////////
-
-            for (var k in sviIndicator) {
-                delete sviIndicator[k];
-            }
-
-            for (var k in scaledSviIndicator) {
-                delete scaledSviIndicator[k];
-            }
-
-            for (var k in weightedSviIndicator) {
-                delete weightedSviIndicator[k];
-            }
-
-            function newSVIvalues(SVIaverage, tempMunic) {
-                sviIndicator[tempMunic] = SVIaverage;
+                var munic4 = la[o].properties[selectedRegion];
+                var theme4 = name;
+                tempString.push(munic4 + '|'+ theme4 +'|'+ tempSum);
```

Comment:
Instead of building strings and then parse them, you might build an array of dictionaries. For instance, in each item of your array, you would have {'zone': 'Lisboa', 'theme': 'Population', 'value': 10.3}